gcc/rust/typecheck/rust-tyty-variance-analysis.cc

```diff
@@ -585,7 +585,7 @@ FieldVisitorCtx::add_constrints_from_param (ParamType &param, Variance variance)
   for (size_t i = type_param_ranges[param_i];
        i < type_param_ranges[param_i + 1]; i++)
     {
-      regions.push_back (parent_regions[i]);
+      regions.push_back (parent_regions[i].value);
```

Comment:
Shouldn't we change the type contained into the vector ? This would allow us to remove the `.value` and keep the type information.